### DISCUSSION TITLE
GATConv error: AssertionError assert self.lin_edge is not None

### DISCUSSION THREAD

**Discussion by JueXiuHuang** [pyg-team/pytorch_geometric#4606] [Q&A] [answered]
Currently I'm using GATConv with edge_attr, but I got the error above.
Below is some information with my code:

```
class GraphAttentionLayer(nn.Module):
    def __init__(self, emb_dim=256, ff_dim=1024):
        super(GraphAttentionLayer, self).__init__()
        self.linear1 = nn.Linear(emb_dim, ff_dim)
        self.linear2 = nn.Linear(ff_dim, emb_dim)
        self.gat = GATConv(emb_dim, emb_dim, heads=8, dropout=0.2, add_self_loops=False)
        
    def forward(self, x, edge_index, edge_attr):
        # x.shape >>> num_node, feat_dim
        # edge_index.shape >>> 2, num_edges
        # edge_attr.shape >>> num_edges, feat_dim2
        x = self.gat(x, edge_index, edge_attr)
        
        return x
```
I've test the shape of x, edge_index and edge_attr, and their shape is the same as the comment in code. And I've try to find what self.lin_edge means in source code, but I didn't find any information about this, how can I solve this problem?
By the way, the code is OK if I don't pass edge_attr into GATConv, so I think this might related with edge_attr.

**Comment by rusty1s** [comment, marked as answer]
Please note that you need to specify the `edge_dim` attribute in `GATConv` if you want to use it in combination with `edge_attr`.

**Comment by MarcPoulin1** [comment]
@rusty1s how do you calculate the edge_dim? For heterogeneous graph neural network  each edge type has it's own edge_dim since edge_dim = number of features?
